### PR TITLE
Add Broker details page

### PR DESCRIPTION
**Pull request by rottencandy**
**Fixes**: 
https://issues.redhat.com/browse/ODC-4994
https://issues.redhat.com/browse/ODC-5062

**Description**: 
Add a details page for Broker.
Add a list page for Triggers.

**Screen shots / Gifs for design review**: 
tmp0
tmp1
@openshift/team-devconsole-ux @invincibleJai 

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/knative-plugin/src/components/eventing/BrokerDetailsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { DetailsPage, ListPage } from '@console/internal/components/factory';
+import { navFactory, Kebab, Page } from '@console/internal/components/utils';
+
+import { DetailsForKind } from '@console/internal/components/default-resource';
+import { EventingBrokerModel, EventingTriggerModel } from '../../models';
+import TriggerList from './triggers-list/TriggerList';
+import { referenceForModel } from '@console/internal/module/k8s';
+// import BrokerTriggers from './BrokerTriggers';
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/components/eventing/BrokerDetailsPage.tsx:9]: nit: remove comments and sort imports

File: frontend/packages/knative-plugin/src/types.ts [hunk 2]
```diff
@@ -85,6 +85,18 @@ export type EventBrokerKind = {
   };
 } & K8sResourceKind;
 
+export enum TriggerConditionTypes {
+  Ready = 'Ready',
+  brokerReady = 'BrokerReady',
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/types.ts:90]: any reason for having diff case here

Comment (rottencandy) [reply frontend/packages/knative-plugin/src/types.ts:90]: This is used in `TriggerRow.tsx`, but I can remove it and use string directly if it isn't required.

File: frontend/packages/knative-plugin/src/components/eventing/BrokerDetailsPage.tsx (hunk 1 above)
Comment (rottencandy) [reply frontend/packages/knative-plugin/src/components/eventing/BrokerDetailsPage.tsx]: Updated.

File: frontend/packages/knative-plugin/src/components/eventing/BrokerDetailsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
```
Comment (invincibleJai) [inline frontend/packages/knative-plugin/src/components/eventing/BrokerDetailsPage.tsx:1]: we may not need dedicated details page here as only new tab is getting contributed and can make use of `HorizontalNavTab` plugin , WDYT?

```
{
    type: 'HorizontalNavTab',
    properties: {
      model:BrokerModel,
      page: {
        name: 'Triggers',
        href: 'triggers',
      },
      loader: () =>
        ...
    },
    flags: {
      required: [flag1],
    },
  },
```

File: frontend/packages/knative-plugin/src/components/eventing/triggers-list/TriggerListPage.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import { referenceForModel } from '@console/internal/module/k8s';
+import { EventingTriggerModel } from '../../../models';
+import TriggerList from './TriggerList';
+
+const TriggerListPage: React.FC<React.ComponentProps<typeof ListPage>> = (props) => (
+  <ListPage
+    canCreate={false}
+    {...props}
+    kind={referenceForModel(EventingTriggerModel)}
+    ListComponent={TriggerList}
```
Comment (sahil143) [inline frontend/packages/knative-plugin/src/components/eventing/triggers-list/TriggerListPage.tsx:12]: you can use the fieldSelector field here to filter our the triggers based on the broker name
```
fieldSelector={`spec.broker=${<broker name>}`}
```

File: frontend/packages/knative-plugin/src/components/eventing/triggers-list/TriggerRow.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  style,
+  customData,
+}) => {
+  const {
+    metadata: { name, namespace, creationTimestamp, uid },
+    spec: { subscriber, filter, broker: connectedBroker },
+  } = obj;
+
+  if (customData?.broker && customData.broker === connectedBroker) {
+    return null;
```
Comment (sahil143) [inline frontend/packages/knative-plugin/src/components/eventing/triggers-list/TriggerRow.tsx:26]: Don't need this if you use the fieldSelector

File: frontend/packages/knative-plugin/src/components/eventing/BrokerDetailsPage.tsx (hunk 3 above)
Comment (rottencandy) [reply frontend/packages/knative-plugin/src/components/eventing/BrokerDetailsPage.tsx]: Updated.

File: frontend/packages/knative-plugin/src/components/eventing/triggers-list/TriggerListPage.tsx (hunk 4 above)
Comment (rottencandy) [reply frontend/packages/knative-plugin/src/components/eventing/triggers-list/TriggerListPage.tsx:17]: `fieldSelector` seems to have issue with `spec.broker`, so will be going with current implementation.
```Error Loading Triggers: field label not supported: spec.broker```

**Comment by invincibleJai** [comment]
thanks @rottencandy looks good, will verify it once

**Comment by bmignano** [comment]
@rottencandy it seems like there might be some weird alignment issues with a few of the columns – other than that small fix it looks good to me!

**Comment by invincibleJai** [comment]
@rottencandy thanks, it looks good. For Channel , Broker list under eventing we may need ready and conditions in column looks like. Let me know if we can handle it it same PR or in a diff PR

**Comment by rottencandy** [comment]
> @rottencandy thanks, it looks good. For Channel , Broker list under eventing we may need ready and conditions in column looks like. Let me know if we can handle it it same PR or in a diff PR

Added in the Channels PR.

**Review by sahil143 — APPROVED** [review]
/lgtm

**Comment by rottencandy** [comment]
Rebased.

**Comment by invincibleJai** [comment]
/lgtm
/approve

Verified the changes , works as expected

**Comment by rottencandy** [comment]
Rebased.

**Pull request merged**